mlflow/projects/docker.py

```diff
@@ -26,18 +27,30 @@
 
 def validate_docker_installation():
     """
-    Verify if Docker is installed on host machine.
+    Verify if Docker is installed and running on host machine.
     """
-    try:
-        docker_path = "docker"
-        process._exec_cmd([docker_path, "--help"], throw_on_error=False)
-    except EnvironmentError:
+    if shutil.which("docker") is None:
         raise ExecutionException(
             "Could not find Docker executable. "
             "Ensure Docker is installed as per the instructions "
             "at https://docs.docker.com/install/overview/."
         )
 
+    cmd = ["docker", "ps"]
```

Comment:
Perhaps a silly question, but could a 'ps' statement be dangerous and create a memory issue if a user has a large number of containers? Since we're using subprocess.communicate() in `mlflow.utils.process`, this in-memory read buffer could potentially hit the default timeout and issue a misleading failure?
Would it be safer to do a fixed-length / size operation of `docker info` to get the sys information from the docker daemon?

Reply:
Thanks for the comment. Whatever can confirm docker daemon is running is fine. Let's use `docker info`.

Reply:
Fixed!